webapp/lib/app/nook/controller_view_helper.dart

```diff
@@ -144,39 +144,27 @@ void _populateReplyPanelView(List<model.SuggestedReply> replies) {
   }
 }
 
-void _populateTagPanelView(List<model.Tag> tags) {
+void _populateTagPanelView(Map<String, List<model.Tag>> tagsByGroup, bool showShortcut, Set<String> tagIds) {
   _view.tagPanelView.clear();
 
-  // Important tags first, then sort by text string
-  tags.sort((t1, t2) {
-    switch (t1.type) {
-      case model.TagType.important:
-        if (t2.type == model.TagType.important) {
-          return t1.text.compareTo(t2.text);
-        } else {
-          return -1;
-        }
-        break;
-      default:
-        if (t2.type == model.TagType.important) {
-          return 1;
-        } else {
-          return t1.text.compareTo(t2.text);
-        }
-    }
-  });
+  var groupNames = (tagsByGroup?.keys ?? []).toList()..sort((a, b) => a.toLowerCase().compareTo(b.toLowerCase()));
 
-  for (var tag in tags) {
-    var tagView = new TagActionView(tag.text, tag.shortcut, tag.tagId, TAG_CONVERSATION_BUTTON_TEXT);
-    if (controller.currentConfig.mandatoryExcludeTagIds != null && controller.currentConfig.mandatoryExcludeTagIds.contains(tag.tagId)) {
-      var questionIcon = SpanElement()..className = "fas fa-info";
-      var tooltip = Tooltip(questionIcon, "Once you've added this tag, you won't be able to see the conversation anymore due to access restrictions on this tag.", position: TooltipPosition.bottom);
-      tooltip.renderElement.classes.add('tag-tooltip');
-      tagView.action.append(tooltip.renderElement);
+  for (String tagGroupName in groupNames) {
+    var tagGroupView = TagGroupView(tagGroupName, tagGroupName, DivElement(), DivElement());
+    var tags = tagsByGroup[tagGroupName];
+    Map<String, TagView> tagViewsById = {};
+    for (model.Tag tag in tags) {
+      tagViewsById[tag.docId] = TagView(tag.text, tag.text, selectable: true, shortcut: showShortcut ? tag.shortcut : "")
+        ..onSelect = () { _view.appController.command(UIAction.addTag, new TagData(tag.tagId)); };
```

Comment:
nit: prefer to use Dart's specific syntax for one-line functions
```suggestion
        ..onSelect = () => _view.appController.command(UIAction.addTag, new TagData(tag.tagId));
```